pkg/catalog/manager/traverse.go

```diff
@@ -243,25 +255,40 @@ func (m *Manager) traverseAndUpdate(helm *helmlib.Helm, commit string, cmt *Cata
 	} else if clusterCatalog != nil {
 		clusterCatalog.Catalog = *catalog
 	}
-	cmt.catalog = catalog
-	cmt.projectCatalog = projectCatalog
-	cmt.clusterCatalog = clusterCatalog
-	if len(terrors) > 0 {
+	/*conditions need to be set here to stop templates from updating
+	each time when they have no changes
+	*/
+	v3.CatalogConditionUpgraded.True(catalog)
+	v3.CatalogConditionDiskCached.True(catalog)
+	if len(createErrors) > 0 {
+		v3.CatalogConditionRefreshed.False(catalog)
+		v3.CatalogConditionRefreshed.Message(catalog, fmt.Sprintf("Error syncing catalog %v", catalog.Name))
+		cmt.catalog = catalog
+		cmt.projectCatalog = projectCatalog
+		cmt.clusterCatalog = clusterCatalog
+		if _, err := m.updateCatalogInfo(cmt, catalogType, "", false, false); err != nil {
+			return err
+		}
+		return errors.Errorf("failed to create templates. Multiple error occurred: %v", createErrors)
+	}
+	if len(updateErrors) > 0 {
+		v3.CatalogConditionRefreshed.False(catalog)
+		v3.CatalogConditionRefreshed.Message(catalog, fmt.Sprintf("Error syncing catalog %v", catalog.Name))
+		cmt.catalog = catalog
+		cmt.projectCatalog = projectCatalog
+		cmt.clusterCatalog = clusterCatalog
 		if _, err := m.updateCatalogInfo(cmt, catalogType, "", false, true); err != nil {
 			return err
 		}
```

Comment:
logic seems comparable here, may it can be put in a function to help with readability?

Reply:
added function as well as unit test for new function